### PR TITLE
Fix events filtering

### LINKED ISSUE #208: Allow new event types
Panel, cocktail session, company visits, etc.

### PR DESCRIPTION
**Pull request by pedronunes19**
Closes #208

### CODE REVIEW

File: resources/js/Pages/Event.vue [hunk 1]
```diff
@@ -125,16 +125,27 @@ const colorPicker = () => {
...
-                            : router.get(route('home') + '#enroll-section') // HACK: this is a hack
-                        : router.get(route('register'))
+                    if (isEnrolled && event.external_url) {
+                    } else {
+                        $page.props.auth.user
+                            ? isEnrolled
+                                ? router.put(route('event.join', event))
+                                : router.get(route('home') + '#enroll-section') // HACK: this is a hack
+                            : router.get(route('register'));
+                    }
```
Comment (toni-santos) [inline resources/js/Pages/Event.vue]: Is this "if" correct?

Comment (pedronunes19) [reply resources/js/Pages/Event.vue]: I meant for the button to have no behavior in that case, so it's just a link essentially

Comment (pedronunes19) [reply resources/js/Pages/Event.vue]: Looks very sketchy though

**Comment by CiscoPr** [comment]
don't forget to rebase

File: resources/js/Pages/Event.vue (hunk 1 above)
Comment (pedrosilva17) [reply resources/js/Pages/Event.vue]: Can't you do this instead? Looks better than an empty if statement imo
```suggestion
                    if (!(isEnrolled && event.external_url)) {
                        $page.props.auth.user
                            ? isEnrolled
                                ? router.put(route('event.join', event))
                                : router.get(route('home') + '#enroll-section') // HACK: this is a hack
                            : router.get(route('register'));
                    }
```

Comment (pedronunes19) [reply resources/js/Pages/Event.vue]: yes that's much better, I don't know why I thought an empty if was a good idea :skull:

**Review by Naapperas — APPROVED** [review]

**Pull request merged**